packages/client/e2e/analytics.spec.ts

```diff
@@ -0,0 +1,145 @@
+      const lastPositionAmount = await analyticsPage
+        .locator("[data-testid='lastPosition']")
+        .textContent()
+
+      const regexp = RegExp(`[-+,.0-9]`, "g")
+      expect(lastPositionAmount).toMatch(regexp)
+    })
+
+    test("Last position amount is updated periodically", async () => {
+      const lastpositionLocator = analyticsPage.locator(
+        "[data-testid='lastPosition']",
+      )
+      const initiallastPositionAmount = await lastpositionLocator.textContent()
+
+      if (initiallastPositionAmount)
+        await expect(lastpositionLocator).not.toContainText(
+          initiallastPositionAmount,
+          { timeout: ElementTimeout.NORMAL },
+        )
+    })
+
+    test("Correct text color is displayed based on position value", async () => {
+      const initialpositionLocator = analyticsPage.locator(
+        "[data-testid='lastPosition']",
+      )
+      const amount = Number(
+        (await initialpositionLocator.textContent())?.replace(/[.,]/g, ""),
+      )
+
+      if (amount < 0) {
+        expect(await initialpositionLocator.getAttribute("color")).toEqual(
+          "negative",
+        )
+      } else
+        expect(await initialpositionLocator.getAttribute("color")).toEqual(
+          "positive",
+        )
+    })
+
+    test.skip("Graph display coherent values", async () => {
+      // TODO
+      // more complex. may requires datatest-id to reliably assert graph content
+    })
+  })
+
+  test.describe("Positions section", () => {
+    test("Position nodes are showing tooltip information for each currencies", async () => {
+      for (const currency of currencies) {
+        const currencyCircle = analyticsPage
+          .locator("g.node")
+          .filter({ hasText: currency })
+          .first()
+        const currencyTooltip = analyticsPage
+          .locator("[data-testid='tooltip']", { hasText: currency })
+          .first()
+
+        await currencyCircle.hover()
+        await expect(currencyTooltip).toBeVisible({
+          timeout: ElementTimeout.AGGRESSIVE,
+        })
+        const regexp = RegExp(`${currency} [-+,.0-9]`, "g")
+        expect(
+          await currencyTooltip.textContent(),
+          `tooltip for ${currency} doesn't match expected pattern`,
+        ).toMatch(regexp)
+      }
+    })
+
+    test("Position nodes can be moved", async () => {
+      const nzdNode = analyticsPage
+        .locator("g.node")
+        .filter({ hasText: "NZD" })
+        .first()
+
+      const jpyNode = analyticsPage
+        .locator("g.node")
+        .filter({ hasText: "JPY" })
+        .first()
+
+      const nzdInitialPosition = await nzdNode.boundingBox()
+      await nzdNode.dragTo(jpyNode)
+      expect(nzdInitialPosition).not.toEqual(await nzdNode.boundingBox())
+    })
+  })
+
+  test.describe("PnL section", () => {
+    test("PnL value is displayed for each currencies", async () => {
+      const pnlSection = analyticsPage
+        .locator("div:last-of-type")
```

Comment:
do we need all the "last-of-type" / last() stuff? (trying to KISS as always)

Reply:
thank for raising, I've updated with a better locator strategy that avoid using css pseudo-class

Reply:
I'm curious now .. what does the Playwright UI say if you select that element?

Reply:
.. and are we going to transition to Page Objects, as I understand them .. we could do this progressively
e.g. this could be replaced by PO.getPnLSection(), I think ?

Reply:
Here is the `pnlSection` selection in Playwright UI
<img width="1484" alt="Screenshot 2023-10-17 at 11 31 12" src="https://github.com/AdaptiveConsulting/ReactiveTraderCloud/assets/103059406/9d753829-2e95-4b8f-b107-dcba2af79132">
when it is chained the locator looks like 
`locator('div').filter({ has: locator('div').filter({ hasText: 'PnL' }) }).last().locator('div').filter({ hasText: 'EURCAD' }).first().getByTestId('priceLabel')`
<img width="1481" alt="Screenshot 2023-10-17 at 11 35 16" src="https://github.com/AdaptiveConsulting/ReactiveTraderCloud/assets/103059406/fbcde27c-b13f-492f-8377-2e91e714d0ea">

Reply:
I agree that we are lacking implementation of Page object pattern.  I suggest that we tackle the whole PO implementation as part of the dedicated ticket to be prioritised shortly